File: java/src/test/java/com/kcl/GetSchemaTypeTest.java
package com.kcl;

import com.kcl.api.API;
import com.kcl.api.Spec.ExecProgramArgs;
import com.kcl.api.Spec.GetSchemaTypeMappingArgs;
import com.kcl.api.Spec.GetSchemaTypeMappingResult;
import com.kcl.api.Spec.KclType;

import org.junit.Assert;
import org.junit.Test;

public class GetSchemaTypeTest {

    private static final String TEST_FILE = "./src/test_data/schema.k";

    @Test
    public void testGetSchemaTypeApi() throws Exception {
        ExecProgramArgs execArgs = ExecProgramArgs.newBuilder().addKFilenameList(TEST_FILE).build();

        GetSchemaTypeMappingArgs args = GetSchemaTypeMappingArgs.newBuilder().setExecArgs(execArgs).build();

        API apiInstance = new API();
        GetSchemaTypeMappingResult result = apiInstance.getSchemaTypeMapping(args);

        KclType appSchemaType = result.getSchemaTypeMappingMap().get("app");
        String replicasType = appSchemaType.getPropertiesOrThrow("replicas").getType();
        Assert.assertEquals("int", replicasType);
    }
}

File: kotlin/src/test/kotlin/com/kcl/GetSchemaTypeTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.execProgramArgs
import com.kcl.api.getSchemaTypeMappingArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class GetSchemaTypeTest {
    companion object {
        private const val TEST_FILE = "./src/test_data/schema.k"
    }

    @Test
    fun testGetSchemaTypeApi() {
        val args = getSchemaTypeMappingArgs { execArgs = execProgramArgs { kFilenameList += TEST_FILE } }
        val api = API()
        val result = api.getSchemaTypeMapping(args)
        val appSchemaType = result.schemaTypeMappingMap["app"] ?: throw AssertionError("App schema type not found")
        val replicasAttr = appSchemaType.properties["replicas"] ?: throw AssertionError("App schema type of `replicas` not found")
        assertEquals(replicasAttr.type, "int")
    }
}
